alibi/explainers/pd_variance.py

```diff
@@ -0,0 +1,660 @@
+            it applies). For categorical features, the corresponding value in the `grid_points` can be
+            specified either as array of strings or array of integers corresponding the label encodings.
+            Note that the label encoding must match the ordering of the values provided in the `categorical_names`.
+
+        Returns
+        -------
+        explanation
+            An `Explanation` object containing the data and the metadata of the calculated partial dependence
+            curves and feature importance/interaction. See usage at `Partial dependence variance examples`_ for details
+
+            .. _Partial dependence variance examples:
+                https://docs.seldon.io/projects/alibi/en/stable/methods/PartialDependenceVariance.html
+        """
+        if method not in Method.__members__.values():
+            raise ValueError(f"Unknown mode. Received ``method={method}``. "
+                             f"Accepted `method` names are: {get_options_string(Method)}")
+
+        # get number of features
+        n_features = X.shape[1]
+
+        # construct features if not provided based on the method
+        if features is None:
+            features = list(range(n_features))
+
+            if method == Method.INTERACTION:
+                features = list(combinations(features, 2))
+
+        # compute partial dependence functions
+        params = {
+            'X': X,
+            'features': features,
+            'percentiles': percentiles,
+            'grid_resolution': grid_resolution,
+            'grid_points': grid_points
+        }
+
+        if not isinstance(self.pd_explainer.predictor, BaseEstimator):
+            params.update({'kind': Kind.AVERAGE.value})  # type: ignore[dict-item]
+
+        # compute partial dependence for each feature
+        pd_explanation = self.pd_explainer.explain(**params)  # type: ignore[arg-type]
+
+        if method == Method.IMPORTANCE:
+            if not all([isinstance(f, numbers.Integral) for f in features]):
+                raise ValueError(f"For ``method='{Method.IMPORTANCE.value}'`` all features must be integers.")
+            # compute feature importance
+            buffers = self._compute_feature_importance(pd_explanation=pd_explanation, features=features)
+        else:
+            if not all([isinstance(fs, tuple) and len(fs) for fs in features]):
+                raise ValueError(f"For ``method='{Method.INTERACTION.value}'`` all features must be "
+                                 f"tuples of length 2.")
+            # compute feature interaction
+            buffers = self._compute_feature_interaction(pd_explanation=pd_explanation,
+                                                        features=features)  # type: ignore[arg-type]
+
+        # update meta and return built explanation
+        self.meta['params'].update(self.pd_explainer.meta['params'])
```

Comment:
there is `kind='average'`. I can remove it, but will be required in the plotting functionality. So the alternative is to add it there

Reply:
I think it would be less confusing if we didn't include it in here as it's not something specified by the user but internal to the algorithm, then we could use it inside the plotting function (is it simple to do or do we need to inject it in the `Explanation` object, plot, and then remove?).